src/game_battlealgorithm.cpp

```diff
@@ -1176,6 +1156,32 @@ void Game_BattleAlgorithm::Skill::Init() {
 			Output::Warning("Algorithm Skill: Invalid skill animation ID {}", skill.animation_id);
 		}
 	}
+
+	// Skill reflect checks
+	// Skills invoked by items ignore reflect
+	if (!item) {
```

Comment:
Ahh my bad. There is an edge case I forgot.

Suppose there are 4 enemies and one of them has reflect, and you decide to cast a spell on the enemy party. Now suppose the enemy with reflect is killed before the spell starts, in that case we should not reflect the spell.

TL;DR Unfortunately I think this means this logic needs to move to `TargetFirst()`. You can make TargetFirst call a virtual function which skill overloads.